src/query/storages/hive/src/hive_catalog.rs

```diff
@@ -81,6 +82,42 @@ impl HiveCatalog {
         Ok(ThriftHiveMetastoreSyncClient::new(i_prot, o_prot))
     }
 
+    pub async fn get_partitions_async(
+        &self,
+        db: String,
+        table: String,
+        partition_names: Vec<String>,
+    ) -> Result<Vec<Partition>> {
+        let client = self.get_client()?;
+        tokio::task::spawn_blocking(move || {
```

Comment:
Why we need `spawn_blocking` here?

Reply:
thrift operator is io-blocking ,may block tokio worker thread